include/z5/common.hxx

```diff
@@ -8,8 +8,8 @@
 // include boost::filesystem or std::filesystem header
 // and define the namespace fs
 #ifdef WITH_BOOST_FS
-    #ifndef BOOST_FILESYSTEM_NO_DEPERECATED
-        #define BOOST_FILESYSTEM_NO_DEPERECATED
+    #ifndef BOOST_FILESYSTEM_NO_DEPRECATED
+        #define BOOST_FILESYSTEM_NO_DEPRECATED
```

Comment:
Is `BOOST_FILESYSTEM_NO_DEPRECATED` still required? I'm asking because `BOOST_FILESYSTEM_NO_DEPRECATED` was clearly not defined until now, and yet it has been working.

BTW, it is clearly a z5 typo, not a Boost typo, because as far as I can see, `BOOST_FILESYSTEM_NO_DEPERECATED` has never been used in Boost:
https://github.com/search?q=org%3Aboostorg+BOOST_FILESYSTEM_NO_DEPERECATED
unlike `BOOST_FILESYSTEM_NO_DEPRECATED`:
https://github.com/search?q=org%3Aboostorg+BOOST_FILESYSTEM_NO_DEPRECATED&type=commits

Reply:
This is probably not required anymore since all major builds are using c++17 and std::filesystem now,
Anyway, good to fix this in case some legacy builds with boost filesystem are still around.

Reply:
Actually `BOOST_FILESYSTEM_NO_DEPRECATED` will disable deprecated part of the Boost API. I understand this is an internal issue that will not result in z5 API changes. Since all tests still pass, I guess it's OK.